pmm-app/src/pmm-qan/panel/components/Filters/hooks/useFilters.ts

```diff
@@ -19,7 +25,25 @@ export const useFilters = (): [any, boolean, boolean] => {
       try {
         const result = await FiltersService.getQueryOverviewFiltersList(labels, from, to, columns[0]);
 
+        const filtersGroups = Object.keys(result)
+          .filter((key) => !FILTERS_GROUPS.some((group) => group.dataKey === key))
+          .filter((key) => !HIDDEN_FILTER_LABELS.includes(key))
+          .reduce((groups, key) => {
+            if (result[key].name) {
+              groups.push({
+                name: key
```

Comment:
Could you extract this to an util function with a descriptive name?

Otherwise looks good.